src/wmnds/patterns/header-v2/_header-v2.js

```diff
@@ -0,0 +1,270 @@
+              targetListItem.querySelector('.wmnds-mega-menu__sub-menu-link').focus();
+              headerEl.classList.add('wmnds-header--mega-menu-submenu-open');
+            } else {
+              targetListItem.classList.remove('open');
+              headerEl.classList.remove('wmnds-header--mega-menu-submenu-open');
+            }
+          });
+        });
+
+        // mobile collapse for third level menus
+        const collapseMenus = menu.querySelectorAll(
+          '.wmnds-mega-menu__sub-menu-item .wmnds-mega-menu__collapse-toggle'
+        );
+        collapseMenus.forEach(collapseToggle => {
+          collapseToggle.addEventListener('click', () => {
+            const panel = collapseToggle.nextElementSibling;
+            collapseToggle.classList.toggle('open');
+            if (panel.style.maxHeight) {
+              panel.style.maxHeight = null;
+            } else {
+              panel.style.maxHeight = `${panel.scrollHeight}px`;
+            }
+          });
+        });
+      }
+    }
+    // end mobile nav function
+
+    // init mobile nav function
+    handleMobileMenu(mobileMenu);
+    mobileMenu.addListener(handleMobileMenu);
+
+    const topLevelLinks = menu.querySelectorAll('.wmnds-mega-menu__primary-menu-link');
+
+    // check if level 3 menu is present, if so add modifier class
+    const hasSubmenuChildren = menu.querySelectorAll('.wmnds-mega-menu__sub-menu-child-menu').length !== 0;
+    if (hasSubmenuChildren) {
+      menu.querySelectorAll('.wmnds-mega-menu__sub-menu').forEach(subMenu => {
+        subMenu.classList.add('wmnds-mega-menu__sub-menu--has-child-menus');
+      });
+    }
+
+    // handle events within each top level list item
+    topLevelLinks.forEach((topLevelLink, topLevelLinkIndex) => {
+      let mousedown = false;
+      // return list item parent of the current link if it exists else return the link
+      const topLevelListItem = topLevelLink.parentNode.tagName === 'LI' ? topLevelLink.parentNode : topLevelLink;
+      const subMenuLinks = topLevelListItem.querySelectorAll('.wmnds-mega-menu__sub-menu-link');
+
+      // if link is focused via click
+      const handleMouseup = () => {
+        mousedown = false;
+        document.removeEventListener('mouseup', handleMouseup);
+      };
+
+      const handleMousedown = () => {
+        mousedown = true;
+        document.addEventListener('mouseup', handleMouseup);
+      };
+
+      // if link is focused via keyboard
+      const handleKeyFocus = () => {
+        topLevelLink.focus();
+        if (!mousedown) {
+          // add keyFocus class for when top level link is focused but megamenu is not active
+          menu.classList.add('keyFocus');
+          topLevelListItem.classList.add('keyFocus');
+        }
+      };
+
+      // remove keyFocus and active classes on blur
+      const handleBlur = () => {
+        menu.classList.remove('keyFocus', 'active');
+        topLevelListItem.classList.remove('keyFocus', 'active');
```

Comment:
Any reference to adding/removing the "keyFocus" class to the topLevelListItem as it is just used for styling, we can work out the correct styling from the class added to the menu.classList.

The class also currently doesn't get removed onBlur so another reason to remove/change the way it works.

Finally I would recommend adding the class: ".wmnds-mega-menu--using-keyFocus" to the main menu item as it makes a bit more sense than just "keyFocus".

You will also need to change the logic a little so that when a user "hovers" that the modifier class for keyFocus is removed from the menu item.